pandas/core/groupby/generic.py

```diff
@@ -1205,9 +1206,17 @@ def first_not_none(values):
             if v is None:
                 return DataFrame()
             elif isinstance(v, NDFrame):
+
+                # this is to silence a FutureWarning
```

Comment:
what case is this hitting?

Reply:
This is raised in `pandas/tests/groupby/test_groupby.py::test_inconsistent_return_type`

A code sample:
```
    df = DataFrame(
        dict(
            A=["Tiger", "Tiger", "Tiger", "Lamb", "Lamb", "Pony", "Pony"],
            B=Series(np.arange(7), dtype="int64"),
            C=date_range("20130101", periods=7),
        )
    )

    def f(grp):
        if grp.name == "Tiger":
            return None
        return grp.iloc[0]

    result = df.groupby("A").apply(f)[["B"]]
```